plugins/hypervisors/xenserver/src/test/java/com/cloud/hypervisor/xenserver/resource/XenServer625ResourceTest.java

```diff
@@ -30,7 +31,7 @@
 
     @Before
     public void beforeTest() {
-        super.citrixResourceBase = new Xenserver625Resource();
+        super.citrixResourceBase = Mockito.spy(new Xenserver625Resource());
```

Comment:
i don't see this spy being used, why create it?

Reply:
Oh, this one... It is because there is a hierarchy of test classes.. :(.. the property `citrixResourceBase` needs to be either a mock or spy object because of the testes executed in the super class. Therefore, needed to create a spy object whenever attributing an object to this property.